### ISSUE TITLE
Interconnected ForLoops

### ISSUE THREAD

**Issue by kmataru** [Wouterdek/NodeNetwork#28]
Would it be possible in the _ExampleCodeGenApp_ project to connect a _Current Index_ endpoint from a _ForLoop_ to a _First/Last Index_ endpoint of other _ForLoop_?

Apparently the Validator won't allow it.

In the short timeframe I made the Input Connection Validator to accept it (Generic Arguments checks) and dynamically get the Observable Value. But then I observed that its Value is _null_.

![image](https://user-images.githubusercontent.com/8168616/53987549-016f1600-412a-11e9-9962-fa9442b3c498.png)

Everything is straightforward but this _Current Index_ is a bit confusing for me at this stage and I was thinking to buzz you. 🤕

What are your thoughts?

**Comment by Wouterdek** [comment]
Jup, that's a bug.

**Comment by kmataru** [comment]
I don't know how much does it helps or if it helps at all, but this is what I did.
(Sure it doesn't cover all the cases)

_`ValueNodeInputViewModel<T>`_

```cs
ConnectionValidator = pending =>
{
    var isValid = pending.Output is ValueNodeOutputViewModel<T>;
    if (!isValid)
    {
        var checkType = typeof(ValueNodeOutputViewModel<T>);
        var pendingOutputType = pending.Output.GetType();

        if (checkType.IsGenericType && pendingOutputType.IsGenericType)
        {
            var checkTypeGenericArguments = checkType.GetGenericArguments();
            var pendingOutputTypeGenericArguments = pendingOutputType.GetGenericArguments();

            if (checkTypeGenericArguments.Length == pendingOutputTypeGenericArguments.Length)
            {
                var areGenericArgumentPairsValid = true;

                for (int idx = 0; idx < checkTypeGenericArguments.Length; idx++)
                {
                    var isAssignableFrom = checkTypeGenericArguments[idx].IsAssignableFrom(pendingOutputTypeGenericArguments[idx]);
                    if (!isAssignableFrom)
                    {
                        areGenericArgumentPairsValid = false;
                        break;
                    }
                }

                if (areGenericArgumentPairsValid)
                {
                    isValid = true;
                }
            }
        }
    }

    return new ConnectionValidationResult(isValid, null);
};
```

_`NodeOutputViewModel`_

```cs
public IObservable<T> GetValue<T>()
{
    return (this as dynamic).Value;
}
```

**Comment by Wouterdek** [comment]
This should be resolved by commit fa8a29b. Feel free to reopen if it is not.
